connectors/agent/service_manager.py

```diff
@@ -59,26 +59,34 @@ async def run(self):
         try:
             while self._running:
                 try:
-                    self._logger.info("Starting connector services")
+                    logger.info("Starting connector services")
+                    config = self._agent_config.get()
                     self._multi_service = get_services(
                         ["schedule", "sync_content", "sync_access_control", "cleanup"],
-                        self._agent_config.get(),
+                        config,
                     )
+                    log_level = config.get("service", {}).get(
+                        "log_level", logging.INFO
+                    )  # Log Level for connectors is managed like this
+                    connectors.logger.set_logger(log_level, filebeat=True)
+                    # Log Level for agent connectors component itself
+                    connectors.agent.logger.update_logger_level(log_level)
```

Comment:
It looks like we're only actually changing our log level based on what we're sent at startup, but I don't see anything in `protocol.py` or `config.py` that actually respects a changed log level later. Am I missing it? I would expect this logic here to be part of the checkin handler.

Reply:
> It looks like we're only actually changing our log level based on what we're sent at startup, but I don't see anything in protocol.py or config.py that actually respects a changed log level later. 

The service manager allows the connector service to be restarted (see the `outer while True` loop) - the connector service will restart and it will take whatever the config was updated to (including the updated log level). This is the same "loop" in which we are getting the updated config with ES credentials upon fleet enrolment. 

See the place we actually trigger the restart (this PR changes the `try_update` logic to also trigger for logging change) https://github.com/elastic/connectors/blob/4f6ca9bb4d9d05f8c400252d717088a7734f4b99/connectors/agent/protocol.py#L72

The try_update takes the checkin payload from [UnitExpected](https://github.com/elastic/elastic-agent-client/blob/86f524ddc40d8aae5a01244ae223fb87f8b33265/elastic-agent-client.proto#L186) protobuf object (that contains both config and log_level)

Reply:
You can modify the agent logging level in the logs tab at the very bottom:
<img width="300" alt="Screenshot 2024-09-24 at 11 39 52" src="https://github.com/user-attachments/assets/88aec4d8-e1c9-4ff3-8f5d-98caaac86221">



I performed 2 syncs: one with debug one with info level, the log level changes work as expected:
<img width="1492" alt="Screenshot 2024-09-24 at 11 58 23" src="https://github.com/user-attachments/assets/5ed1cf48-1779-4df2-93e6-7b32fb39f1d6">

Reply:
It somehow worked before (was agent process setting some env variable for us?) .... I detect the log level changes explicitly as a part of this PR: https://github.com/elastic/python-elastic-agent-client/pull/73

Reply:
I performed a couple of syncs, the python component logs with expected level (see variance in info, debug logs). In the last interval I set level to `error` and we see no logs:

<img width="839" alt="Screenshot 2024-09-24 at 12 37 24" src="https://github.com/user-attachments/assets/661ab1b8-ffd4-4863-882b-e8db8bfc0915">

Reply:
> the connector service will restart and it will take whatever the config was updated to (including the updated log level)

I wouldn't expect a restart to be required to pick up a log level change, when running through Agent. I know that's needed with connectors today, because we only read from config.yml once on startup, but for this, I'd expect a checkin to change the log level, but not interrupt anything in the process or syncs. Am I misunderstanding? Regardless, that's out of the scope of this PR, so not blocking, but let's be sure to revisit.

> I performed a couple of syncs, the python component logs with expected level (see variance in info, debug logs). In the last interval I set level to error and we see no logs

Fantastic! Thanks for verifying.